src/course/domain/repository/course-repository.ts

```diff
@@ -0,0 +1,18 @@
+import { Course } from '../course';
+import { CourseFilters } from '../course-filters';
+import { CourseId } from '../course-id';
+
+export interface CourseRepository {
+  create(course: Course): Promise<Course>;
+  getAll(
+    page?: number,
+    limit?: number,
+    filters?: CourseFilters
+  ): Promise<{
+    courses: Course[];
+    total: number;
+  }>;
+  getById(id: CourseId): Promise<Course | null>;
+  delete(id: CourseId): Promise<void>;
+  edit(id: CourseId, course: Partial<Course>): Promise<Course>;
```

Comment:
The edit method signature uses Partial<Course>, which leaks domain entities/value objects into the repository/update API. This couples infrastructure adapters to domain implementation details and makes it awkward to map persistence DTOs to domain objects cleanly.

**Sugerencia:**
```typescript
Define an UpdateCourse DTO or a dedicated type for updatable fields (primitive types or simple DTOs) that repository/adapters accept, then map to domain objects in the adapter. E.g.:
export interface UpdateCourseDTO { title?: string; description?: string; instructor?: string; duration?: string; level?: string; }
Then change signature to edit(id: CourseId, data: UpdateCourseDTO): Promise<Course>.
```